lang/cs/Org.Apache.REEF.Tang/Implementations/ClassHierarchy/AvroClassHierarchySerializer.cs

```diff
@@ -0,0 +1,484 @@
+﻿/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using Microsoft.Hadoop.Avro;
+using Microsoft.Hadoop.Avro.Container;
+using Newtonsoft.Json;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Exceptions;
+using Org.Apache.REEF.Tang.Implementations.ClassHierarchy.AvroDataContract;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Types;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Tang.Implementations.ClassHierarchy
+{
+    /// <summary>
+    /// AvroClassHierarchySerializer is to serialize and deserialize ClassHierarchy
+    /// </summary>
+    public class AvroClassHierarchySerializer : IClassHierarchySerializer
+    {
+        private static readonly Logger LOGGER = Logger.GetLogger(typeof(AvroClassHierarchySerializer));
+
+        /// <summary>
+        /// Default constructor for the interface IClassHierarchySerializer
+        /// </summary>
+        [Inject]
+        public AvroClassHierarchySerializer()
+        {
+        }
+
+        /// <summary>
+        /// Avro schema
+        /// </summary>
+        /// <returns></returns>
+        public string GetSchema()
```

Comment:
Not really. It is used in testing code to make sure Hadoop.Avro is working properly.